### PR TITLE
kv get with revision

### PR DESCRIPTION
**Pull request by scottf**
https://github.com/nats-io/nats-architecture-and-design/issues/93

**KeyValue.java** 
* Added interface method overload which takes a revision parameter

**NatsJetStreamManagement.java** 
* Changed imports for readability
* Underlying MessageInfo object changed to be consistent, so had to manually throw on error

**NatsKeyValue.java**
* Implemented get with revision with refactor to original get to share common code.

* NatsKeyValueWatchSubscription.java
* Use minor refactor 

**KeyValueTests.java**
* test the implementation

### CODE REVIEW

**Review by edgefrog — APPROVED** [review]

File: src/main/java/io/nats/client/KeyValue.java [hunk 1]
```diff
@@ -45,6 +45,19 @@
...
+     * @param key the key
+     * @param revision the revision
+     * @return the KvEntry object or null if not found.
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data
+     * @throws IllegalArgumentException the server is not JetStream enabled
+     */
+    KeyValueEntry get(String key, long revision) throws IOException, JetStreamApiException;
+
```
[the remark this answers is not in the document]
Comment (scottf) [reply src/main/java/io/nats/client/KeyValue.java:60]: Any day now...

File: src/main/java/io/nats/client/impl/NatsJetStreamManagement.java [hunk 2]
```diff
@@ -218,7 +216,7 @@ public ConsumerInfo getConsumerInfo(String streamName, String consumer) throws I
     public MessageInfo getMessage(String streamName, long seq) throws IOException, JetStreamApiException {
         validateNotNull(streamName, "Stream Name");
         String subj = String.format(JSAPI_MSG_GET, streamName);
-        Message resp = makeRequestResponseRequired(subj, seqBytes(seq), jso.getRequestTimeout());
+        Message resp = makeRequestResponseRequired(subj, MessageGetRequest.seqBytes(seq), jso.getRequestTimeout());
```
[the remark this answers is not in the document]
Comment (scottf) [reply src/main/java/io/nats/client/impl/NatsJetStreamManagement.java:219]: I usually use static imports but this time I felt the code self documented better

File: src/main/java/io/nats/client/impl/NatsJetStreamManagement.java [hunk 3]
```diff
@@ -230,8 +228,8 @@ public MessageInfo getLastMessage(String streamName, String subject) throws IOEx
         validateNotNull(streamName, "Stream Name");
         validateSubject(subject, true);
         String getSubject = String.format(JSAPI_MSG_GET, streamName);
-        Message resp = makeRequestResponseRequired(getSubject, lastBySubjectBytes(subject), jso.getRequestTimeout());
-        return new MessageInfo(resp);
+        Message resp = makeRequestResponseRequired(getSubject, MessageGetRequest.lastBySubjectBytes(subject), jso.getRequestTimeout());
+        return new MessageInfo(resp).throwOnHasError();
```
[the remark this answers is not in the document]
Comment (scottf) [reply src/main/java/io/nats/client/impl/NatsJetStreamManagement.java:232]: To be more consistent with other method implementations .

File: src/main/java/io/nats/client/impl/NatsKeyValue.java [hunk 4]
```diff
@@ -84,18 +84,40 @@ String getStreamSubject() {
      */
     @Override
     public KeyValueEntry get(String key) throws IOException, JetStreamApiException {
-        return getLastMessage(validateNonWildcardKvKeyRequired(key));
+        return _kvGetLastMessage(validateNonWildcardKvKeyRequired(key));
```
[the remark this answers is not in the document]
Comment (scottf) [reply src/main/java/io/nats/client/impl/NatsKeyValue.java:87]: Doesn't really mean anything, it's just a convention I use for delegate methods.

**Review by edgefrog — APPROVED** [review]
Ok

**Review by jnmoyne — APPROVED** [review]
LGTM

**Pull request merged**